Source/Events.Store.MongoDB/Processing/Filters/FilterDefinitionRepository.cs

```diff
@@ -54,7 +54,7 @@ public async Task PersistFilter(IPersistableFilterDefinition filterDefinition, C
                 var document = filterDefinition switch
                     {
                         TypeFilterWithEventSourcePartitionDefinition definition => definition.ToStoreRepresentation(),
-                        _ => new FilterDefinition(filterDefinition.TargetStream, filterDefinition.SourceStream)
+                        _ => new FilterDefinition(filterDefinition.TargetStream, filterDefinition.SourceStream, filterDefinition.Partitioned)
```

Comment:
See previous comment - the logic here that the default here is `FilterDefinition` doesn't quite match up with that the _default_ in the core is `IPersistableFilterDefinition`.